src/cpp/fastdds/domain/DomainParticipantFactory.cpp

```diff
@@ -68,6 +77,28 @@ static void set_qos_from_attributes(
     qos.transport().listen_socket_buffer_size = attr.listenSocketBufferSize;
     qos.name() = attr.getName();
     qos.flow_controllers() = attr.flow_controllers;
+
+    if (merge_properties)
+    {
+        for (auto property : attr.properties.properties())
+        {
+            // If the property from attr does not exist in the qos, then append it; if it does, then the XML one takes preference
```

Comment:
I have clarified the comment removing any reference to XML. The intention as follows:

* If the Attributes property is not in the QoS, then it should be appended.
* If the Attributes property is also in the QoS, the the value in the QoS is overridden with that of the Attributes.

Reply:
> A test checking the overriding behavior would be nice to have to ensure the expected behavior is the one documented.

As disscussed internally, no further tests will be added since this functionality is already tested in `DDSStatistics.discovery_topic_physical_data_user_defined_xml`